pkg/server/span_stats.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package server
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/rpc"
+	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/gogo/protobuf/jsonpb"
+	gwruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"google.golang.org/grpc"
+	"io"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+var _ serverpb.SpanStatsServer = &spanStatsServer{}
+
+type spanStatsServer struct {
+	server *Server
+}
+
+func (s *spanStatsServer) SetSpanBoundaries(ctx context.Context, request *serverpb.SetSpanBoundariesRequest) (*serverpb.SetSpanBoundariesResponse, error) {
+	s.server.node.stores.VisitStores(func(st *kvserver.Store) error {
+		st.SetBucketBoundaries()
+		return nil
+	})
+	return &serverpb.SetSpanBoundariesResponse{}, nil
+}
+
+func (s *spanStatsServer) RegisterService(g *grpc.Server) {
+	serverpb.RegisterSpanStatsServer(g, s)
+}
+
+// RegisterGateway starts the gateway (i.e. reverse proxy) that proxies HTTP requests
+// to the appropriate gRPC endpoints.
+func (s *spanStatsServer) RegisterGateway(
+	ctx context.Context, mux *gwruntime.ServeMux, conn *grpc.ClientConn,
+) error {
+	return serverpb.RegisterSpanStatsHandler(ctx, mux, conn)
+}
+
+func (s *spanStatsServer) dialNode(ctx context.Context, nodeID roachpb.NodeID) (serverpb.SpanStatsClient, error) {
+	addr, err := s.server.gossip.GetNodeIDAddress(nodeID)
+	if err != nil {
+		return nil, err
+	}
+	log.Infof(ctx, "dial node %d with addr: %s", nodeID, addr.String())
+	rpcConnection := s.server.rpcContext.GRPCDialNode(addr.String(), nodeID, rpc.DefaultClass)
+	log.Infof(ctx, "Have connection for node %d", nodeID)
```

Comment:
Since you asked me to be nitpicky: we also generally avoid capitalizing (and adding periods) for logging since it doesn't compose -- it's too sentencey for what is otherwise a fragment of info in a larger log file.